src/Sylius/Bundle/RbacBundle/spec/Security/RbacVoterSpec.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Bundle\RbacBundle\Security;
+
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Rbac\Model\IdentityInterface;
+use Sylius\Component\Rbac\Authorization\PermissionMapInterface;
+use Sylius\Component\Rbac\Provider\PermissionProviderInterface;
+use Sylius\Component\Rbac\Resolver\RolesResolverInterface;
+use Sylius\Component\Rbac\Model\RoleInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\VoterInterface;
+use Doctrine\Common\Cache\Cache;
+
+/**
+ * @author Christian Daguerre <christian@daguer.re>
+ */
+class RbacVoterSpec extends ObjectBehavior
+{
+    function let(
+        PermissionMapInterface $permissionMap,
+        PermissionProviderInterface $permissionProvider,
+        RolesResolverInterface $rolesResolver,
+        Cache $cache
```

Comment:
As I said before, extracting caching can simplify this voter, in case of specs it will be much easier to write specification from caching and voting logic standalone.